cpp/core_pyos.h

```diff
@@ -32,20 +37,61 @@ class ReadError {
 };
 
 inline Str* GetUserName(int uid) {
-  assert(0);
+  Str *result = kEmptyString;
+
+  if (passwd *pw = getpwuid(uid))
+  {
+    result = new Str(pw->pw_name);
+  }
+
+  return result;
 }
 
 inline Str* OsType() {
-  // uname()[0].lower()
-  return new Str("TODO");
+  Str *result = kEmptyString; // TODO(Jesse): Should we set this to "unknown" or something?
```

Comment:
I think we should raise an `errno` exception in all these cases.  But I'm not sure if there is a good example of that in the C++ code.  I would make it `assert(0)` or `assert(!"TODO")` for now